src/helpers/query/assignments.js

```diff
@@ -1086,60 +1086,72 @@ export const assignmentFetchAll = async (adminId, orgType, orgId, includeScores
   const axiosInstance = getAxiosInstance();
 
   const queries = districts.map(async (district) => {
+    const filters = [
+      {
+        fieldFilter: {
+          field: { fieldPath: 'districts' },
+          op: 'ARRAY_CONTAINS',
+          value: { stringValue: district.id },
+        },
+      },
+      {
+        compositeFilter: {
+          op: 'OR',
+          filters: [
+            {
+              fieldFilter: {
+                field: { fieldPath: 'name' },
+                op: 'EQUAL',
+                value: { stringValue: name },
+              },
+            },
+            {
+              fieldFilter: {
+                field: { fieldPath: 'normalizedName' },
+                op: 'EQUAL',
+                value: { stringValue: normalizedName },
+              },
+            },
+          ],
+        },
+      },
+    ];
+
+    if (adminId) {
+      filters.push({
+        fieldFilter: {
+          field: { fieldPath: '__name__' },
+          op: 'NOT_EQUAL',
+          value: { referenceValue: `${getBaseDocumentPath()}/${FIRESTORE_COLLECTIONS.ADMINISTRATIONS}/${adminId}` },
+        },
+      });
+    }
+
     const requestBody = {
       structuredQuery: {
         from: [{ collectionId: FIRESTORE_COLLECTIONS.ADMINISTRATIONS }],
         where: {
           compositeFilter: {
             op: 'AND',
-            filters: [
-              {
-                fieldFilter: {
-                  field: { fieldPath: 'districts' },
-                  op: 'ARRAY_CONTAINS',
-                  value: { stringValue: district.id },
-                },
-              },
-              {
-                compositeFilter: {
-                  op: 'OR',
-                  filters: [
-                    {
-                      fieldFilter: {
-                        field: { fieldPath: 'name' },
-                        op: 'EQUAL',
-                        value: { stringValue: name },
-                      },
-                    },
-                    {
-                      fieldFilter: {
-                        field: { fieldPath: 'normalizedName' },
-                        op: 'EQUAL',
-                        value: { stringValue: normalizedName },
-                      },
-                    },
-                  ],
-                },
-              },
-            ],
+            filters,
           },
         },
       },
     };
 
     try {
       const response = await axiosInstance.post(`${getBaseDocumentPath()}:runQuery`, requestBody);
-      return response.data.filter((data) => data.document);
+      return mapFields(response.data);
     } catch (error) {
       console.error('Error fetching assignment by name: ', error);
       return null;
     }
   });
 
   const results = await Promise.all(queries);
-  return Array.isArray(results) ? results.flat() : null;
+  return Array.isArray(results) ? results.flat() : [];
```

Comment:
We should actually let the error throw instead of returning an empty array so that TanStack knows that an error happened. We can then handle the error in the onError callback.